include/revng/ADT/Concepts.h

```diff
@@ -21,9 +21,8 @@ concept equality_comparable = requires(T &&LHS, T &&RHS) {
   { LHS == RHS } -> std::convertible_to<bool>;
 };
 
-// clang-format off
```

Comment:
Did we get rid of all of them?
If so, please ban `clang-format off` in `rcc.yml`

Reply:
I envision problems if we ban `clang-format off`. Let me explain why.

In `revng-c` we're gonna use MLIR, and MLIR requires certain headers to be included in a specific order (isn't that the case @drblallo @andrealinux1 ?).

In our `clang-format` configuration we force the reordering of headers, and that would break the ordering required from MLIR. So at the moment @andrealinux1 had to use `clang-format off` at least around those nasty `#include`s. This means that either we find a way to solve this include ordering problem, or it's already NOGO to ban `clang-format off` because we already know that we're gonna need it at least there.

There's of course the option of explicitly changing `check-conventions` so that `clang-fomat off` is allowed only in the specific MLIR files, but then `check-conventions` will be "polluted" by filenames, potentially causing MLIR-related changesets in `revng-c` to require changes in `revng` too. It can be done, but I'd honestly prefer to avoid this solution.

Reply:
I got read of what I could. But some pain-points are still there:

- chaining `requires` in the concept definition sucks. `clang-format` will definitely fix it, but who knows when (There's still no option for it, even in trunk).
- when separate statements inside `requires` block grows beyond certain size, it gets really ugly.
- Stuff like this:
```
using rp_error = std::variant<std::monostate,
                              rp_simple_error,
                              rp_document_error>;
```
- there's a macro in `Assert.h` that look really broken when formatted.

Reply:
Can you post a `diff` of things you don't like when you drop all of `clang-format off`?

Reply:
(moved to mattermost, but keeping the thread open as a reminder)

Reply:
done :tada: